File: frontend/src/types/product.ts
export interface ProductImage {
  id: number;
  image_url: string;
  sort_order: number;
}

export interface ProductAlias {
  id: number;
  alias: string;
}

export interface ProductVariant {
  id: number;
  slug: string;
  variant_name: string;
  price: string;
  compare_at_price: string | null;
  availability_status: string;
  delivery_eta: string | null;
  stock_quantity: number;
  is_box_damaged: boolean;
  is_active: boolean;
}

export interface Product {
  id: number;
  name: string;
  slug: string;
  category: string;
  subcategory: string | null;
  series: string;
  product_number: string | null;
  short_description: string;
  rarity: string;
  is_new: boolean;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  images: ProductImage[];
  aliases: ProductAlias[];
  variants: ProductVariant[];
}

File: backend/app/schemas/product.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class ProductImageRead(BaseModel):
    id: int
    image_url: str
    sort_order: int

    model_config = ConfigDict(from_attributes=True)


class ProductAliasRead(BaseModel):
    id: int
    alias: str

    model_config = ConfigDict(from_attributes=True)


class ProductRead(BaseModel):
    id: int
    name: str
    slug: str
    category: str
    subcategory: str | None = None
    series: str
    product_number: str | None = None
    price: Decimal
    short_description: str
    rarity: str
    availability_status: str
    delivery_eta: str | None = None
    stock_quantity: int
    is_new: bool
    is_active: bool
    created_at: datetime
    updated_at: datetime
    images: list[ProductImageRead] = []
    aliases: list[ProductAliasRead] = []

    model_config = ConfigDict(from_attributes=True)